src/ring_theory/finiteness.lean

```diff
@@ -260,10 +274,34 @@ end
 /-- If `f : A →ₐ[R] B` is surjective with finitely generated kernel and `A` is finitely presented,
 then so is `B`. -/
 lemma of_surjective {f : A →ₐ[R] B} (hf : function.surjective f) (hker : f.to_ring_hom.ker.fg)
-  (hfp : finitely_presented R A) : finitely_presented R B :=
+  (hfp : finite_presentation R A) : finite_presentation R B :=
 equiv (quotient hker hfp) (ideal.quotient_ker_alg_equiv_of_surjective hf)
 
-end finitely_presented
+/-- An algebra is finitely presented if and only if it is a quotient of a polynomial ring whose
+variables are indexed by a fintype by a finitely generated ideal. -/
+lemma iff_quotient_mv_polynomial' : finite_presentation R A ↔ ∃ (ι : Type u_2) [fintype ι]
+  (f : (_root_.mv_polynomial ι R) →ₐ[R] A), (surjective f) ∧ f.to_ring_hom.ker.fg :=
+begin
+  split,
+  { rintro ⟨n, f, hf⟩,
+    set ulift_var := mv_polynomial.alg_equiv_of_equiv R equiv.ulift,
+    refine ⟨ulift (fin n), infer_instance, f.comp ulift_var.to_alg_hom,
+      hf.1.comp ulift_var.surjective,
+      submodule.fg_ker_ring_hom_comp _ _ _ hf.2 ulift_var.surjective⟩,
```

Comment:
Could unpack hypotheses one more level, to avoid `.1` and `.2`:
```suggestion
  { rintro ⟨n, f, hfs, hfk⟩,
    set ulift_var := mv_polynomial.alg_equiv_of_equiv R equiv.ulift,
    refine ⟨ulift (fin n), infer_instance, f.comp ulift_var.to_alg_hom,
      hfs.comp ulift_var.surjective,
      submodule.fg_ker_ring_hom_comp _ _ _ hfk ulift_var.surjective⟩,
```
Same for the other split case